core/src/transaction.rs

```diff
@@ -84,71 +81,107 @@ impl CellOutPoint {
     pub fn cell_key(&self) -> CellKey {
         CellKey::calculate(&self.tx_hash, self.index)
     }
-}
 
-#[derive(Clone, Default, Serialize, Deserialize, Eq, PartialEq, Hash)]
-pub struct OutPoint {
-    pub cell: Option<CellOutPoint>,
-    pub block_hash: Option<H256>,
+    /// Convert from dep group data
+    pub fn from_group_data(data: &[u8]) -> Result<OutPoint, usize> {
+        if data.len() != mem::size_of::<H256>() + mem::size_of::<u32>() {
+            Err(data.len())
+        } else {
+            let tx_hash_start = 0;
+            let tx_hash_end = tx_hash_start + mem::size_of::<H256>();
+            let index_start = tx_hash_end;
+            let index_end = index_start + mem::size_of::<u32>();
+            let tx_hash = H256::from_slice(&data[tx_hash_start..tx_hash_end])
+                .expect("Invalid tx hash length");
+            let mut index_bytes = [0u8; mem::size_of::<u32>()];
+            index_bytes.copy_from_slice(&data[index_start..index_end]);
+            // Deserialize as little endian u32
+            let index = u32::from_le_bytes(index_bytes);
+            Ok(OutPoint::new(tx_hash, index))
+        }
+    }
+
+    /// Convert OutPoint to one dep group data item
+    pub fn to_group_data(&self) -> Vec<u8> {
+        [self.tx_hash.to_vec(), self.index.to_le_bytes().to_vec()].concat()
+    }
```

Comment:
Please add some tests for the from/to conversion.

Reply:
Changed to CellKey's implementation (see the code above in the same file).